skimage/feature/_texture.pyx

```diff
@@ -12,24 +12,38 @@ from .._shared.transform cimport integrate
 cdef extern from "numpy/npy_math.h":
     double NAN "NPY_NAN"
 
-
-def _glcm_loop(cnp.uint8_t[:, ::1] image, double[:] distances,
+ctypedef fused any_int:
+    cnp.uint8_t
+    cnp.uint16_t
+    cnp.uint32_t
+    cnp.uint64_t
+    cnp.int8_t
+    cnp.int16_t
+    cnp.int32_t
+    cnp.int64_t
+        
+def _glcm_loop(any_int[:, ::1] image, double[:] distances,
                double[:] angles, Py_ssize_t levels,
                cnp.uint32_t[:, :, :, ::1] out):
     """Perform co-occurrence matrix accumulation.
 
     Parameters
     ----------
     image : ndarray
-        Input image, which is converted to the uint8 data type.
+        Integer typed input image. Only positive valued images are supported. 
+        If type is other than uint8, the argument `levels` needs to be set.
     distances : ndarray
         List of pixel pair distance offsets.
     angles : ndarray
         List of pixel pair angles in radians.
     levels : int
-        The input image should contain integers in [0, levels-1],
+        The input image should contain integers in [0, `levels`-1],
```

Comment:
This is correct and intuitive, since you can't index with negative values. (One could imagine an elaborate scheme with negative indices, but that is not really worth thinking about!) The signed type support is handy because I imagine that many people unwittingly end up with int64 images.